src/context-parser.js

```diff
@@ -180,6 +180,14 @@ FastParser.prototype.walk = function(i, input, endsWithEOF) {
             if(this.tags[0].toLowerCase() === this.tags[1].toLowerCase()) {
                 reconsume = 0;  /* see 12.2.4.13 - switch state for the following case, otherwise, reconsume. */
                 this.matchEndTagWithStartTag(symbol);
+                /* 
+                  After matchEndTagWithStartTag (with start tag name == end tag name), 
+                  the state will be transition back to DATA. 
```

Comment:
This approach seems simpler and easier to follow along. And, not doing a reset for https://github.com/yukinying/context-parser/blob/reset-end-tag/src/context-parser.js#L239 and https://github.com/yukinying/context-parser/blob/reset-end-tag/src/context-parser.js#L242 means we'll need to have another place to reset tags[0] for those cases, which seems complicated.  thoughts?